fairlearn/metrics/_group_feature.py

```diff
@@ -47,18 +47,21 @@ class GroupFeature:
 
     def __init__(self, base_name: str, feature_vector, index: int, name: str | None = None):
         """Help with the metrics."""
-        self.classes_ = np.unique(feature_vector)
-        self.raw_feature_ = feature_vector
+        nw_feature_vector = nw.from_native(feature_vector, pass_through=True, allow_series=True)
+        is_nw_series = isinstance(nw_feature_vector, nw.Series)
+
+        self.classes_ = np.unique(nw_feature_vector)
```

Comment:
I noticed that `self.classes_` is set but unused. If commented out, the only failing tests are those that specifically test for the attribute value

Reply:
It's entirely possible that it was simply there because I thought I'd use it. Since `GroupFeature` is not public-facing, I've not objection to removing it if unused.

Reply:
Thanks for the feedback and the information @riedgar-ms - I removed it and adjusted the tests that were checking for such attribute in https://github.com/fairlearn/fairlearn/pull/1546/commits/5e001b016b76e6c7c151a5ffb5aee9216c890622 and https://github.com/fairlearn/fairlearn/pull/1546/commits/131a084fc8bf428411511ab3a5709f4f721d89ff

It should be easy and straightforward to revert those changes in case you change your mind.

A compromise is to have it as a property, so that the computation of unique values is triggered _only_ if and when required. Let me know if this option is preferable